src/model/wallet.ts

```diff
@@ -727,29 +908,35 @@ const migrateSecrets = async () => {
     }
 
     logger.sentry('Got secret, now idenfifying wallet type');
-    const type = identifyWalletType(seedPhrase);
+    const type = identifyWalletType(seedphrase);
     logger.sentry('Got type: ', type);
-    let hdnode, node, existingAccount;
+    let hdnode: undefined | ethers.utils.HDNode.HDNode,
+      node: undefined | ethers.utils.HDNode.HDNode,
+      existingAccount: undefined | Wallet;
     switch (type) {
-      case WalletTypes.privateKey:
-        existingAccount = new ethers.Wallet(seedPhrase);
+      case EthereumWalletType.privateKey:
+        existingAccount = new ethers.Wallet(seedphrase);
         break;
-      case WalletTypes.mnemonic:
-        hdnode = ethers.utils.HDNode.fromMnemonic(seedPhrase);
+      case EthereumWalletType.mnemonic:
+        hdnode = ethers.utils.HDNode.fromMnemonic(seedphrase);
         break;
-      case WalletTypes.seed:
-        hdnode = ethers.utils.HDNode.fromSeed(seedPhrase);
+      case EthereumWalletType.seed:
+        hdnode = ethers.utils.HDNode.fromSeed(seedphrase);
         break;
       default:
     }
 
-    if (!existingAccount) {
+    if (!existingAccount && hdnode) {
       logger.sentry('No existing account, so we have to derive it');
       node = hdnode.derivePath(`${DEFAULT_HD_PATH}/0`);
       existingAccount = new ethers.Wallet(node.privateKey);
       logger.sentry('Got existing account');
     }
 
+    if (!existingAccount) {
+      return;
```

Comment:
technically should we return null here

Reply:
oh I guess we haven't typed this particular function's signature

Reply:
Oops. Just added it.